src/DeferredResponseFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App;
+
+use Nyholm\Psr7\Response;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+class DeferredResponseFactory implements ResponseFactoryInterface
+{
+    private StreamFactoryInterface $streamFactory;
+
+    public function __construct(StreamFactoryInterface $streamFactory)
+    {
+        $this->streamFactory = $streamFactory;
+    }
+
+    public function createResponse(int $code = 200, string $reasonPhrase = '', $data = null): ResponseInterface
+    {
+        return new DeferredResponse($data, $this->createResponseInternal($code, $reasonPhrase), $this->streamFactory);
+    }
+
+    private function createResponseInternal(int $code = 200, string $reasonPhrase = ''): ResponseInterface
+    {
+        if (2 > \func_num_args()) {
```

Comment:
How does that work? There are exactly two arguments for `createResponseInternal()` passed so the check should never be true.

Reply:
This a copy of the code from Psr17Factory

Reply:
I don't understand it.